sql/core.go

```diff
@@ -217,6 +217,11 @@ type Replacer interface {
 	Inserter
 }
 
+type Updater interface {
+	// Update the given row. Provides both the old and new rows.
```

Comment:
Probably should describe any special errors implementors are expected to return, if any.

Reply:
No special errors are expected to be returned. `Delete` is the only one that returns a special error.